command/resource/client/grpc-client.go

```diff
@@ -31,9 +32,30 @@ func NewGRPCClient(config *GRPCConfig) (*GRPCClient, error) {
 }
 
 func dial(c *GRPCConfig) (*grpc.ClientConn, error) {
-	// TODO: decide if we use TLS mode based on the config
-	dialOpts := []grpc.DialOption{
-		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	err := checkCertificates(c)
+	if err != nil {
+		return nil, err
+	}
+	var dialOpts []grpc.DialOption
+	if c.GRPCTLS {
+		tlsConfig, err := SetupTLSConfig(c)
+		if err != nil {
+			return nil, fmt.Errorf("failed to setup tls config when tried to establish grpc call: %w", err)
+		}
+		dialOpts = append(dialOpts, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
+	} else {
+		dialOpts = append(dialOpts, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	}
+
 	return grpc.Dial(c.Address, dialOpts...)
 }
+
+func checkCertificates(c *GRPCConfig) error {
+	if c.GRPCTLS {
+		if (c.CertFile != "" && c.KeyFile == "") || (c.CertFile == "" && c.KeyFile != "") {
```

Comment:
I think this is missing the case where the certfile is not empty but the keyfile is empty.

Maybe we can break this conditional to make it easier to understand, something along those lines:
```golang
certFileEmpty := (c.CertFile == "")
keyFileEmpty := (c.CertFile == "")

// both files need to be empty or both files need to be provided
if certFileEmpty != keyFileEmpty {
...
```

Reply:
sounds good